core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java

```diff
@@ -140,7 +141,8 @@ private PluginsLoadResult loadPlugins() {
     pluginsToLoadByKey.putAll(globalConfig.getEmbeddedPluginPathsByKey());
     Set<Path> plugins = new HashSet<>(pluginsToLoadByKey.values());
 
-    var config = new Configuration(plugins, globalConfig.getEnabledLanguages(), Optional.ofNullable(globalConfig.getNodeJsVersion()), globalConfig.getAllowedPluginIds());
+    var config = new Configuration(plugins, globalConfig.getEnabledLanguages(), Optional.ofNullable(globalConfig.getNodeJsVersion()),
+      DataflowBugDetection.getPluginAllowList(globalConfig.isDataflowBugDetectionEnabled()));
```

Comment:
I'm tempted to go one step further and pass the `isDataflowBugDetectionEnabled` boolean to the PluginLoader configuration (like enabledLanguages) and only internally decide which plugins to load.